src/components/BaseSelect.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { OptionInfo } from '@/types/option';
+
+/**
+ * セレクトボックスの情報の型
+ *
+ * @property {string} className class名
+ * @property {OptionInfo[]} optionList 選択肢
+ * @property {{(e: React.ChangeEvent<HTMLSelectElement>) => void}} handleFunc ハンドラメソッド
+ */
+type Props = {
+    className?: string;
+    optionList: OptionInfo[];
+    handleFunc: (e: React.ChangeEvent<HTMLSelectElement>) => void;
+}
+
+/**
+ * セレクトボックス表示用コンポーネント
+ *
+ * @param {Props} props セレクトボックスの情報
+ * @returns {JSX.Element}
+ */
+export const BaseSelect = (props: Props) => {
+    const { className, optionList, handleFunc } = props;
+
+    return (
+        <select className={className} onChange={handleFunc}>
+            <option value=""></option>
+            {optionList.map((option, index) => {
+                return (
+                    <option key={index} value={option.id}>{option.value}</option>
+                )
+            })}
```

Comment:
DOMの途中でmapメソッドなどを利用する場合は `return` は省略する方が好まれる傾向にあります。

**プロジェクトに方針があればそれに従うのが良いですが、**
省略する記法を支持する理由の1つとして、`{` から始まるブロックを記述すると、その中にロジックを "書けて"しまうことがあげられます。DOMの途中にロジックは極力書かない方が良いので書けないようにしておこうということですね。

```suggestion
            {optionList.map((option, index) => (
                <option key={index} value={option.id}>{option.value}</option>
            ))}
```

Reply:
mapメソッドからreturnを削除いたしました。
4f9055c47ac095539f9c4048b7357c45559cad8e